File: New BackEnd/demo/src/main/java/com/example/demo/models/Customer.java
package com.example.demo.models;
import com.example.demo.embedded.Person;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;
import java.util.List;
import java.util.UUID;

@Entity
@Data
@Table(name = "customer")
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private UUID id;
    private String email;
    private String phone;
    @Embedded
    private Person generaleInfo;
    @OneToMany(mappedBy = "customer")
    private List<Construction> constructions;
    @ManyToOne
    private Company company;
}

File: photovoltaicV2/src/app/services/models/customer.ts
/* tslint:disable */
/* eslint-disable */
import { Company } from '../models/company';
import { Construction } from '../models/construction';
import { Person } from '../models/person';
export interface Customer {
  company?: Company;
  constructions?: Array<Construction>;
  email?: string;
  generaleInfo?: Person;
  id?: number;
  phone?: string;
}
